### DISCUSSION TITLE
Need Urgent Assistance: Cronjob Setup for Firefly III in Portainer Docker Compose File

### DISCUSSION THREAD

**Discussion by thorrrr** [firefly-iii/firefly-iii#7243] [Firefly III]
Hello 
Been trying for weeks to get the cron job running with my docker compose file.
Yes i am a noob but i have read the docs several times and had great help from Elements/Matrix channel.
But was asked to open a ticket on here to see if i could get help to see what it is not running.

Here is my compose file redacted see [here](https://i.imgur.com/Dv9Oahs.png)
Please note if i uncomment the env_file i get errors.
Under the current files everything starts but the cron does not run see [here](https://i.imgur.com/65yX9RV.png)

Log from Cron  1 line "crond: crond (busybox 1.35.0) started, log level 8"
Log from main app see [here](https://logs.notifiarr.com/?22486fe797611ffc#GpcCcxHC3EqJTg3ybXyB4a7b9PRJ5DhFGEitLsfniv5Y)

I have no idea why it shows running or how to get it running and test it.
Please can somebody help me

**Comment by JC5** [comment]
So the error you get when you start **with** the `.env` file in place is different, right? What error is that?

**Comment by thorrrr** [comment]
failed to deploy a stack: open /data/compose/52/.env: no such file or directory

**Reply by JC5** [reply]
[Check out this guide](https://www.portainer.io/blog/using-env-files-in-stacks-with-portainer)

**Reply by thorrrr** [reply]
OK i have read it but still no idea what i need to do.
1. Your docs make no mention about creating a file called env_file and where is that stored and what is in it?
2. Yours docs make no mention about using this functionality or this button hitting this it asks for a file !
3. In the Docs we have multiple paragraphs 

- The Docker image does not support cron jobs.

1. Static cron token
2. Call the cron job from outside the Docker container
3. Call the cron job from the host system
4. Run an image that calls the cron job
5. Expand the docker-compose file

My friend who has used Portainer for 4 years says he cannot fully understand the docs and how your docs and Portainers match up?
Sorry but it is all very confusing  for a new user:(

From the link you sent me 

![image](https://user-images.githubusercontent.com/8170673/225614828-df716e94-6413-45e9-a035-825353832083.png)

**Reply by JC5** [reply]
Sorry, this is also something for you to find out, if the .env file doesn't work then perhaps you can enter the variables manually?

**Comment by lennardk** [comment]
Hi,
> Here is my compose file redacted see [here](https://i.imgur.com/Dv9Oahs.png) Please note if i uncomment the env_file i get errors. Under the current files everything starts but the cron does not run see [here](https://i.imgur.com/65yX9RV.png)
> 
> Log from Cron 1 line "crond: crond (busybox 1.35.0) started, log level 8"
> 
> I have no idea why it shows running or how to get it running and test it. Please can somebody help me
Do you have any more output from the cronjob other than the first line? For example, using this docker-compose file:
```yaml
version: '3.3'

services:
  cron:
    image: alpine
    command: sh -c "echo \"* * * * * wget -qO- https://SOME-URL/\" | crontab - && crond -f -L /dev/stdout"
```

I get the following output for my `cron` container:
```
cron_1  | crond: crond (busybox 1.35.0) started, log level 8
cron_1  | crond: USER root pid   9 cmd wget -qO- https://SOME-URL/
cron_1  | <WEBPAGE>
cron_1  | crond: USER root pid  11 cmd wget -qO- https://SOME-URL/
cron_1  | <WEBPAGE>
cron_1  | crond: USER root pid  13 cmd wget -qO- https://SOME-URL/
cron_1  | <WEBPAGE>
cron_1  | crond: USER root pid  15 cmd wget -qO- https://SOME-URL/
cron_1  | <WEBPAGE>
```

I changed my cron pattern to `* * * * *` to get output every minute. Your pattern `0 3 * * *` means: 03:00 (3 am) every day. Since this in a container, this probably is 03:00 in timezone UTC.

If you have no other output after you have waited for 03:00 to come around, something is probably wrong within the cron container.

**Reply by JC5** [reply]
For what it's worth, this is my docker compose file and it ran perfectly last night:

```
  cron:
    image: alpine
    command: sh -c "echo \"0 3 * * * wget -qO- http://app:8080/api/v1/cron/TOKEN\" | crontab - && crond -f -L /dev/stdout"
```

Notice the backslashes in the command.

Output:

```
crond: crond (busybox 1.35.0) started, log level 8
crond: USER root pid   9 cmd wget -qO- http://app:8080/api/v1/cron/TOKEN
{"recurring_transactions":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Recurring transactions cron job fired successfully."},"auto_budgets":{"job_fired":true,"job_succeeded":true,"job_errored":false,"message":"Auto-budget cron job fired successfully."}}crond: USER root pid  10 cmd wget -qO- http://app:8080/api/v1/cron/jDTURbeIPXTMy8LTaEmLSNSa9NfPrv9q
```
